### ISSUE TITLE
Expired metadata and invalid port

### ISSUE THREAD

**Issue by spuder** [the-maldridge/resinstack#1]
Trying out resinstack and getting this error

`Metadata for targets/riyaz expired`

and

` invalid port ":18.06.0-ce-dind" after host`

```
make img/nomad.qcow2
linuxkit build -format qcow2-bios -name nomad -dir img/ components/kernel.yml components/sysctl.yml components/dhcp.yml components/metadata.yml components/emissary.yml components/acpi.yml components/persist-disk.yml components/docker.yml components/nomad/nomad.yml components/nomad/server.yml components/coredns/coredns.yml components/consul/consul.yml
Building LinuxKit image mkimage to generate output formats
Extract kernel image: linuxkit/kernel:4.9.39
WARN[0005] targets metadata is nearing expiry, you should re-sign the role metadata
WARN[0005] targets/justin metadata is nearing expiry, you should re-sign the role metadata
Pull image: docker.io/linuxkit/kernel:4.9.39@sha256:a409547705b00f4dbc6ba09ad67ae3d2c87e5a7a8d8955e310b834b2220b77b0
Add init containers:
Process init image: linuxkit/init:v0.6
WARN[0017] targets metadata is nearing expiry, you should re-sign the role metadata
Pull image: docker.io/linuxkit/init:v0.6@sha256:33054954194d72f8fb1e446dd725a837ed73ccc091b3c0b6a4fdfa32b117f9b8
Process init image: linuxkit/runc:v0.6
ERRO[0024] Metadata for targets/riyaz expired
WARN[0024] Error getting targets/riyaz: targets/riyaz expired at Mon May 25 15:32:34 -0700 2020
WARN[0024] targets/justin metadata is nearing expiry, you should re-sign the role metadata
WARN[0024] targets metadata is nearing expiry, you should re-sign the role metadata
Pull image: docker.io/linuxkit/runc:v0.6@sha256:582dd80f93b457ee9aeec7d67c18f008d2de9a1929fc5b461acbbb9d47d4d2be
Add onboot containers:
  Create OCI config for linuxkit/mkimage:v0.6
ERRO[0030] Metadata for targets/riyaz expired
WARN[0030] Error getting targets/riyaz: targets/riyaz expired at Mon May 25 14:47:53 -0700 2020
WARN[0030] targets/ian metadata is nearing expiry, you should re-sign the role metadata
WARN[0030] targets metadata is nearing expiry, you should re-sign the role metadata
WARN[0030] targets/justin metadata is nearing expiry, you should re-sign the role metadata
Pull image: docker.io/linuxkit/mkimage:v0.6@sha256:34a0c17ba8c9f97b77927b3111f5f35727622141683220ba6b171876e77a7d7f
  Create OCI config for linuxkit/poweroff:afe8f7dd0d47a7991c54519b0f09124cb8c4e300
ERRO[0037] Metadata for targets/riyaz expired
WARN[0037] Error getting targets/riyaz: targets/riyaz expired at Mon May 25 14:57:45 -0700 2020
WARN[0037] targets metadata is nearing expiry, you should re-sign the role metadata
WARN[0037] targets/ian metadata is nearing expiry, you should re-sign the role metadata
Pull image: docker.io/linuxkit/poweroff:afe8f7dd0d47a7991c54519b0f09124cb8c4e300@sha256:6928333dd3cec070d08722b22bc54a754a05959d240c83b1ee0bb0748b6bf652
FATA[0044] Invalid config: extract service image reference: parse "dummy://docker:18.06.0-ce-dind": invalid port ":18.06.0-ce-dind" after host
make: *** [img/nomad.qcow2] Error 1
```

**Comment by the-maldridge** [comment]
Interesting.  The metadata warning is a well-known issue in linuxkit and can be safely ignored.  The invalid port warning on the other hand is interesting, and I am not sure why you would be getting that.

I will investigate and see what I can come up with.

**Comment by the-maldridge** [comment]
To help debug, can you provide information about your build environment?  OS/version/linuxkit build ref.

**Comment by spuder** [comment]
Running 

OSX 10.14.6
linuxkit version v0.6 (installed with brew)

I changed the image by wrapping it in quotes and it seems to be getting further. Perhaps its a yaml parsing issue?  

before
`image: consul:1.7.2`
after
`image: "consul:1.7.2"`


```
---
services:
  - name: consul
    image: "consul:1.7.2"
```

**Comment by the-maldridge** [comment]
Interesting.  I built from master and cannot duplicate the issue you have described.

**Comment by spuder** [comment]
Since you fixed issue #2 this mysteriously isn't an issue anymore.

**Comment by the-maldridge** [comment]
Mysterious bugs are the worst kind.  Please let me know if it re-occurs.
